lib/facter/facts/aix/disks.rb

```diff
@@ -4,13 +4,28 @@ module Facts
   module Aix
     class Disks
       FACT_NAME = 'disks'
+      ALIASES = %w[blockdevices blockdevice_.*_size'].freeze
 
       def call_the_resolver
+        facts = []
         disks = Facter::Resolvers::Aix::Disks.resolve(:disks)
 
         disks = disks&.empty? ? nil : disks
+        blockdevices = disks&.keys&.join(',')
```

Comment:
why don't you return a resolved fact with value nil if it's nil or empty in order to avoid checking twice if it's nil (once at line 13 and twice here with &)? This is valid for the other disk facts as well.

Reply:
You are suggestion something like
```
return Facter::ResolvedFact.new(FACT_NAME, disks) if disks.nil?
```

Reply:
yes